src/main/drivers/gyro_sync.c

```diff
@@ -36,17 +36,32 @@ uint32_t gyroSetSampleRate(gyroDev_t *gyro, uint8_t lpf, uint8_t gyroSyncDenomin
             gyro->gyroRateKHz = GYRO_RATE_32_kHz;
             gyroSamplePeriod = 31.5f;
         } else {
-#ifdef USE_ACCGYRO_BMI160
-            gyro->gyroRateKHz = GYRO_RATE_3200_Hz;
-            gyroSamplePeriod = 312.0f;
-#else
-            gyro->gyroRateKHz = GYRO_RATE_8_kHz;
-            gyroSamplePeriod = 125.0f;
-#endif
+            switch (gyro->mpuDetectionResult.sensor) {
+            case BMI_160_SPI:
+                gyro->gyroRateKHz = GYRO_RATE_3200_Hz;
+                gyroSamplePeriod = 312.0f;
+                break;
+            case ICM_20649_SPI:
+                gyro->gyroRateKHz = GYRO_RATE_9_kHz;
+                gyroSamplePeriod = 111.1f;
+                break;
+            default:
+                gyro->gyroRateKHz = GYRO_RATE_8_kHz;
+                gyroSamplePeriod = 125.0f;
+                break;
+            }
         }
     } else {
-        gyro->gyroRateKHz = GYRO_RATE_1_kHz;
-        gyroSamplePeriod = 1000.0f;
+        switch (gyro->mpuDetectionResult.sensor) {
+        case ICM_20649_SPI:
+            gyro->gyroRateKHz = GYRO_RATE_1100_Hz;
+            gyroSamplePeriod = 909.1f;
```

Comment:
Again, might as well make this exact.

Reply:
I'm not quite sure what you're asking for on these last couple comments. The division results in a repeating decimal, do you just want me to drop a digit?

Reply:
I'm just asking you to not do the truncation here, but do the exact calculation. I know it's truncated when it is eventually returned, but keeping the numbers exact until they are actually truncated is good practice and more readable. So, I mean:
`gyroSamplePeriod = 1000000.0f / 1100.0f;` and `samplingTime = 1.0f / 9000.0f;`

Reply:
Alright, that makes sense. Thanks for clarifying.